File: coprocessor-and-floating-point/leibniz.py
def leibniz(m):
    resultado = 0
    for i in range (m):
        numerador = (-1) ** i
        division = numerador / (2*i + 1)
        resultado += division
    return resultado * 4

m = int(input("Pon el valor de m: "))
pi = leibniz(m)
print(pi)

File: coprocessor-and-floating-point/Leibniz.hs
module Leibniz where

leibniz :: Int -> Double
leibniz m = 4 * sum [((-1) ** fromIntegral i) / (2 * fromIntegral i + 1) | i <- [0..(m-1)]]

main :: IO ()
main = do
    putStr "Pon el valor de m: "
    input <- getLine
    let m = read input :: Int
    let pi = leibniz m
    print pi